code/object/objectdock.cpp

```diff
@@ -836,7 +844,11 @@ void dock_calc_total_moi_helper(object* objp, dock_function_info* infop)
 	// So for each part:
 
 	// We invert the inverse MOI to get an MOI in the local frame
-	vm_inverse_matrix(&local_moi, &objp->phys_info.I_body_inv);
+	if (!vm_inverse_matrix(&local_moi, &objp->phys_info.I_body_inv)) {
+		// This is done on purpose to indicate a zero inv_moi
+		infop->maintained_variables.matrix_value->a1d[0] = NAN;
+		return;
+	}
```

Comment:
As an alternative, you could set `maintained_variables.bool_value`.  Not saying you should, just be aware of it if you hadn't considered it.

Returning NAN makes mathematical sense but is riskier from an implementation point of view.  However your additional comment takes care of that.

Reply:
I think I'll keep the NaN return from the helper function, since that's NaN's purpose after all, but I'll change the dock_calc_total_moi to instead catch the NaN and return a bool, to make it a bit more straightforward, if anyone else ends up needing it.

Reply:
Okay, but keep the comment that says the `dest` matrix may be NAN if the function returns false.